wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireInput.kt

```diff
@@ -16,183 +16,86 @@
 package com.squareup.wire.gradle
 
 import com.squareup.wire.gradle.WireExtension.ProtoRootSet
+import com.squareup.wire.schema.Location
 import java.io.EOFException
 import java.io.File
 import java.io.RandomAccessFile
-import java.net.URI
-import org.gradle.api.Project
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.Dependency
-import org.gradle.api.artifacts.DependencySet
-import org.gradle.api.artifacts.FileCollectionDependency
-import org.gradle.api.file.SourceDirectorySet
-import org.gradle.api.internal.catalog.DelegatingProjectDependency
-import org.gradle.api.internal.file.FileOrUriNotationConverter
-import org.gradle.api.provider.Provider
+import org.gradle.api.internal.file.FileOperations
 
-/**
- * Builds Wire's inputs (expressed as [InputLocation] lists) from Gradle's objects (expressed as
- * directory trees, jars, and coordinates). This includes registering dependencies with the project
- * so they can be resolved for us.
- */
-internal class WireInput(var configuration: Configuration) {
-  val name: String
-    get() = configuration.name
-
-  private val dependencyFilters = mutableMapOf<Dependency, Collection<WireExtension.Filter>>()
-
-  val dependencies: DependencySet
-    get() = configuration.dependencies
-
-  // Deferred dependency evaluation.
-  val inputFiles = mutableSetOf<File>()
-
-  fun addPaths(project: Project, paths: Set<String>) {
-    for (path in paths) {
-      val dependency = resolveDependency(project, path)
-      configuration.dependencies.add(dependency)
+internal val List<ProtoRootSet>.inputLocations
+  get() = flatMap { rootSet ->
+    rootSet.roots.map { file ->
+      rootSet.inputLocation(file)
     }
   }
 
-  fun addJars(project: Project, jars: Set<ProtoRootSet>) {
-    for (jar in jars) {
-      val unresolvedDependency = jar.srcJar
-        ?: jar.srcProject
-        ?: jar.srcJarAsExternalModuleDependency
-        ?: jar.srcProjectDependency
-        ?: continue
-
-      val dependency = resolveDependency(project, unresolvedDependency)
-      dependencyFilters[dependency] = jar.filters
-      configuration.dependencies.add(dependency)
-    }
+private fun ProtoRootSet.inputLocation(file: File): InputLocation {
+  val includes = when {
+    includes.isEmpty() && excludes.isEmpty() -> listOf("**/*.proto")
+    else -> includes
   }
+  // We store [file] relative to the [project] in order to not invalidate the cache when we don't
+  // have to.
+  return InputLocation(project.relativePath(file.path), includes, excludes)
```

Comment:
Note: It's technically ok to move the directory around as long as its contents stay the same so there might still be spurious cache invalidations here.

Reply:
Remote caching is the big thing I wanna preserve. I think we get that as long as the paths are relative to the project.